cpp/include/cudf/io/text/data_chunk_source_factories.hpp

```diff
@@ -25,6 +26,14 @@
 
 namespace cudf::io::text {
 
+/**
+ * @brief Creates a data source capable of producing device-buffered views of a datasource.
+ * @param data the datasource to be exposed as a data chunk source
+ * @return the data chunk source for the provided datasource. It must not outlive the datasource
+ *         used to construct it.
+ */
+std::unique_ptr<data_chunk_source> make_source(datasource& data);
```

Comment:
Should we document when this overload should be used? AFAIK, using `datasource` is often much slower than directly using a `istream_data_chunk_reader`.

Reply:
Not sure - data_source can use kvikio/cuFile, which we don't yet use in the data_chunk_source. Should I add it to the benchmark to put some numbers to this question?

Reply:
Yes, yes I should

|        T        | size_approx    | GPU Time  |Encoded file size |
|-----------------|------------|-------|------------------------|
|            file | 2^10 = 1024 |  44.450 ms |         999.000 B |
|            file | 2^12 = 4096 |  44.404 ms |         3.944 KiB |
|            file |          2^14 = 16384 |  44.271 ms |        15.718 KiB |
|            file |      2^16 = 65536 |  44.338 ms |        63.220 KiB |
|            file |     2^18 = 262144 |  44.442 ms |       251.584 KiB |
|            file |2^20 = 1048576 |   44.780 ms |         1004.521 KiB |
|            file |   2^22 = 4194304 |   45.677 ms |            3.927 MiB |
|            file |  2^24 = 16777216 |   49.339 ms |            15.726 MiB |
|            file |  2^26 = 67108864 |    68.670 ms |           62.926 MiB |
|            file | 2^28 = 268435456 |   126.198 ms |          251.709 MiB |
|            file |2^30 = 1073741824 |   357.786 ms |       1006.638 MiB |
| file_datasource |      2^10 = 1024 |    1.063 ms |               999.000 B |
| file_datasource |      2^12 = 4096 |     1.063 ms |           3.944 KiB |
| file_datasource |     2^14 = 16384 |      1.072 ms |          15.718 KiB |
| file_datasource |     2^16 = 65536 |     1.088 ms |           63.220 KiB |
| file_datasource |    2^18 = 262144 |      1.147 ms |           251.584 KiB |
| file_datasource |   2^20 = 1048576 |      1.587 ms |          1004.521 KiB |
| file_datasource |   2^22 = 4194304 |   3.879 ms |               3.927 MiB |
| file_datasource |  2^24 = 16777216 |   15.695 ms |           15.726 MiB |
| file_datasource |  2^26 = 67108864 |   65.186 ms |          62.926 MiB |
| file_datasource | 2^28 = 268435456 |  110.538 ms |         251.709 MiB |
| file_datasource |2^30 = 1073741824 |  288.055 ms |        1006.638 MiB |

Reply:
man, that thread took an unexpected turn :D

Reply:
For posterity: This is the difference between an `mmap`ped file and `ifstream`